cl/corpus_importer/signals.py

```diff
@@ -0,0 +1,122 @@
+from functools import partial
+
+from django.conf import settings
+from django.db import transaction
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+
+from cl.corpus_importer.tasks import make_docket_by_iquery
+from cl.lib.redis_utils import (
+    acquire_atomic_redis_lock,
+    get_redis_interface,
+    make_update_pacer_case_id_key,
+    release_atomic_redis_lock,
+)
+from cl.search.models import Court, Docket
+
+
+def update_latest_case_id_and_schedule_iquery_sweep(docket: Docket) -> None:
+    """Updates the latest PACER case ID and schedules iquery retrieval tasks.
+
+    :param docket: The incoming Docket instance.
+    :return: None
+    """
+
+    r = get_redis_interface("CACHE")
+    court_id = docket.court.pk
+    # Get the latest pacer_case_id from Redis using a lock to avoid race conditions
+    # when getting and updating it.
+    update_lock_key = make_update_pacer_case_id_key(court_id)
+    # ttl one hour.
+    lock_value = acquire_atomic_redis_lock(r, update_lock_key, 60 * 60 * 1000)
+
+    current_iquery_pacer_case_id_final = int(
+        r.hget("iquery_pacer_case_id_final", court_id) or 0
+    )
+    iquery_pacer_case_id_status = int(
+        r.hget("iquery_pacer_case_id_status", court_id) or 0
```

Comment:
I think a better name for this one is `iquery_pacer_case_id_current` since it's not the status, it's the current value, right?

Reply:
Sure changed!